### DISCUSSION TITLE
DTI Trace image

### DISCUSSION THREAD

**Discussion by mbrzus** [dipy/dipy#2611] [Preprocessing]
Hello,

I am having a problem with my DTI preprocessing pipeline. I am working with the dataset in which I have 3 types of images from the scanner:
- DWI (4D data with bvals and bvecs)
- TRACEW
- ADC.

I was trying to use DIPY to denoise the 4D diffusion data (using Patch2Self) and then extract the ADC and the TRACE images using dipy dti reconstruction.

# Problem:
The Trace image extracted by DIPY appears to match the ADC image I got from the scanner. Additionally, I can not find any way to match the TRACEW image from the scanner. 

| DIPY TRACE | Scanner ADC | Scanner TRACEW |
| ----------- | ----------- | ----------- |
| ![dipyTrace](https://user-images.githubusercontent.com/44876465/171734463-8890f3e6-7c77-47cb-8586-4a8fa6e7d82a.png) | ![scannerADC](https://user-images.githubusercontent.com/44876465/171734543-4798a843-eedd-4196-8972-d507ddc78623.png) | 
![scannerTrace](https://user-images.githubusercontent.com/44876465/171734598-1297d2a2-d944-4cca-8976-a9e141417caa.png) |

It seems that the scanner images match the information I found [here](https://mriquestions.com/trace-vs-adc-map.html). Additionally, I tried to access the 3x3x3 tensor matrix for each voxel to try to write my own functions to try to match the theory from the article but I was only able to access 2d (3x3) matrix for each voxel. 

# Code:
`dwi = <PATH_TO_IMAGE>
data, affine = load_nifti(dwi)
bvals, bvecs = read_bvals_bvecs(dwi.replace("denoised.nii.gz", ".bval"),
                                dwi.replace("denoised.nii.gz", ".bvec"))
gtab = gradient_table(bvals, bvecs)
maskdata, mask = median_otsu(data, vol_idx=range(0, 26), median_radius=3,
                             numpass=1, autocrop=False, dilate=2)

dtimodel = dti.TensorModel(gtab)
dtifit = dtimodel.fit(maskdata, mask=mask)
Trace = dtifit.trace`

Can anyone explain the differences between the scanner, theory and dipy understanding of the TRACE image?

**Comment by ShreyasFadnavis** [comment]
Hi @mbrzus,

AFAIK `trace` is the sum of the eigenvalues of the diffusion tensor. 

So it should be:
trace = $\lambda_1 + \lambda_2 +\lambda_3$

Not sure what is causing this discrepancy though...
